lib/WatchmanFileCache.js

```diff
@@ -0,0 +1,122 @@
+import fbWatchman from 'fb-watchman';
+
+const SUBSCRIPTION_NAME = 'import-js-subscription';
+
+const CACHE = new Set();
+
+// Flag to quickly check if the cache is enabled.
+let enabled = false;
+
+function subscribe({ client, fbWatch, relativePath }): Promise {
+  const subscription = {
+    // Match any `.js` file
+    expression: ['allof', ['match', '*.js*']],
```

Comment:
re `type`, we might also need to include symlinks. https://facebook.github.io/watchman/docs/expr/type.html